src/torrent_info.cpp

```diff
@@ -665,23 +665,30 @@ namespace {
 			// as long as this file already exists
 			// increase the counter
 			std::string filename = m_files.file_path(i);
-			if (!files.insert(filename).second)
-			{
-				std::string base = remove_extension(filename);
-				std::string ext = extension(filename);
-				do
-				{
-					++cnt;
-					char new_ext[50];
-					std::snprintf(new_ext, sizeof(new_ext), ".%d%s", cnt, ext.c_str());
-					filename = base + new_ext;
-				}
-				while (!files.insert(filename).second);
+			auto const ret = files.insert({filename, i});
+			if (ret.second) continue;
+			// pad files are allowed to collide with each-other, as long as they have
+			// the same size.
+			file_index_t const other_idx = ret.first->second;
```

Comment:
There needs to be a check for `other_idx == -1` before calling `file_flags` etc.

Reply:
ah, right! I'll add a test for that too